### PR TITLE
add e2e test for single cluster jobset

### PR DESCRIPTION
**Pull request by kannon92**
#### What type of PR is this?
/kind cleanup

#### What this PR does / why we need it:
Kueue and JobSet are like two peas in a pod. We should have e2e tests verifying the behavior in single-cluster like we have for multikueue.
#### Which issue(s) this PR fixes:

Fixes #
Partially https://github.com/kubernetes-sigs/kueue/issues/1463

#### Does this PR introduce a user-facing change?

```release-note
NONE
```

### CODE REVIEW

File: test/e2e/singlecluster/e2e_test.go [hunk 1]
```diff
@@ -490,6 +493,85 @@ var _ = ginkgo.Describe("Kueue", func() {
...
+				Obj()
+
+			ginkgo.By("Creating the jobSet", func() {
+				gomega.Expect(k8sClient.Create(ctx, jobSet)).Should(gomega.Succeed())
+			})
+
+			createdLeaderWorkload := &kueue.Workload{}
+			wlLookupKey := types.NamespacedName{Name: workloadjobset.GetWorkloadNameForJobSet(jobSet.Name, jobSet.UID), Namespace: ns.Name}
+
+			ginkgo.By("Waiting for the jobSet to get status updates", func() {
```
Comment (mimowo) [inline test/e2e/singlecluster/e2e_test.go:546]: TBH, I don't think this step is necessary for single cluster, this has very little to do with JobSet. I would drop this step and reduce the pod runtime to `1ms`.

File: test/e2e/singlecluster/e2e_test.go [hunk 2]
```diff
@@ -490,6 +493,85 @@ var _ = ginkgo.Describe("Kueue", func() {
...
+				Queue("main").
+				ReplicatedJobs(
+					testingjobset.ReplicatedJobRequirements{
+						Name:        "replicated-job-1",
+						Replicas:    2,
+						Parallelism: 2,
+						Completions: 2,
+						Image:       "gcr.io/k8s-staging-perf-tests/sleep:v0.1.0",
+						// Give it the time to be observed Active in the live status update step.
+						Args: []string{"5s"},
```
Comment (mimowo) [inline test/e2e/singlecluster/e2e_test.go:532]: I would suggest to use 1ms container runtime. I think it is enough to wait for the JobSet to be finished in this test. For multicluster we verify updates, but live updates to the status are a feature on its own.

File: Makefile [hunk 3]
```diff
@@ -188,14 +191,14 @@ MULTIKUEUE-E2E_TARGETS := $(addprefix run-test-multikueue-e2e-,${E2E_K8S_VERSION
 .PHONY: test-e2e-all
 test-e2e-all: ginkgo $(E2E_TARGETS) $(MULTIKUEUE-E2E_TARGETS)
 
+
```
[the remark this answers is not in the document]
Comment (kannon92) [reply Makefile]: done.

File: hack/e2e-test.sh [hunk 4]
```diff
@@ -22,6 +22,10 @@ SOURCE_DIR="$(cd "$(dirname -- "${BASH_SOURCE[0]}")" && pwd -P)"
 ROOT_DIR="$SOURCE_DIR/.."
 export E2E_TEST_IMAGE=gcr.io/k8s-staging-perf-tests/sleep:v0.1.0
 
+export JOBSET_MANIFEST=https://github.com/kubernetes-sigs/jobset/releases/download/${JOBSET_VERSION}/manifests.yaml
```
[the remark this answers is not in the document]
Comment (kannon92) [reply hack/e2e-test.sh]: updated.

File: hack/e2e-test.sh [hunk 5]
```diff
@@ -52,6 +56,9 @@ function kind_load {
         docker pull $E2E_TEST_IMAGE
 	cluster_kind_load $KIND_CLUSTER_NAME
     fi
+    docker pull registry.k8s.io/jobset/jobset:$JOBSET_VERSION
+    kubectl apply --server-side -f ${JOBSET_CRDS}/*
```
[the remark this answers is not in the document]
Comment (kannon92) [reply hack/e2e-test.sh]: yea I think you are right.

File: test/e2e/singlecluster/e2e_test.go [hunk 6]
```diff
@@ -490,6 +493,85 @@ var _ = ginkgo.Describe("Kueue", func() {
 			})
 		})
 	})
+	ginkgo.When("Creating a JobSet", func() {
```
[the remark this answers is not in the document]
Comment (kannon92) [reply test/e2e/singlecluster/e2e_test.go]: done.

File: test/e2e/singlecluster/e2e_test.go (hunk 1 above)
Comment (kannon92) [reply test/e2e/singlecluster/e2e_test.go]: pushed up a fix.

File: test/e2e/singlecluster/e2e_test.go (hunk 2 above)
Comment (kannon92) [reply test/e2e/singlecluster/e2e_test.go]: thanks for pointing that out. done.

**Review by trasc — COMMENTED** [review]
/lgtm

**Review by mimowo — COMMENTED** [review]
/approve

**Review by alculquicondor — COMMENTED** [review]
/approve

**Pull request merged**